sensu/plugins/check-ceph-usage.rb

```diff
@@ -134,17 +134,25 @@ def run_cmd(cmd)
   def run
     result = run_cmd('ceph df --format=json')
     data = JSON.parse(result)
-    used_percentage = data['stats']['total_used_bytes'] * 100.0 / data['stats']['total_bytes']
-    
-    output = '%RAW Used: ' + used_percentage.to_s
-    output = output + '  ' + result if config[:verbose]
-    
-    if used_percentage >= config[:threshold]
-      warning output if config[:criticality] == 'warning'
-      critical output
-    end
-    
+    pool_usage = Hash.new
+    output = ''
+	
+    data['pools'].each do | pool |
+      usage = pool['stats']['bytes_used'] * 100.0 /(pool['stats']['bytes_used'] + pool['stats']['max_avail'])
+      pool_usage[pool['name']] = usage
+      output = output + 'Pool ' + pool['name'] + " %Used: " + usage.to_s + "\n"
+    end unless result.to_s == ''
+    output = output + result if config[:verbose]
+
+    pool_usage.each do |pool_name, percentage|
+      if percentage >= config[:threshold]
+        warning output if config[:criticality] == 'warning'
+        critical output
```

Comment:
Won't this report only the first pool with a critical level? What if multiple pools are in this state? Wouldn't it be better to report in aggregate, or leave this module as a single pool reporter and create multiple checks, one for each pool?